shaal/DrupalPod
# Apply patch or clone only in uninitiated workspace

# The Problem/Issue/Bug
Fixes https://github.com/shaal/DrupalPod/issues/17 - Gitpod Start tasks are running again when restarting a workspace

## How this PR Solves The Problem
Saving a file `/workspace/gitpod_initiated.status` to make sure DrupalPod apply patches or clone projects only once per workspace, which will avoid overriding existing work.

## Manual Testing Instructions
1. Find an issue on drupal.org that includes a patch.
1. Use DrupalPod browser extension to select that patch.
1. Wait until environment finished running all tasks.
1. Click on Hamburger menu on top-left of VScode interface
1. Click on `Gitpod: Stop Workspace`
1. Wait for workspace to shutdown
1. Click on `Open workspace` to bring it up again
1. Confirm that terminal doesn't display any warnings on patches being reapplied.

Alternative test -
(Instead of shutdown a workspace) -
1. Click on Hamburger menu
1. Click on `Gitpod: Share Workspace Snapshot`
1. Copy to clipboard the link Gitpod generated
1. Paste the link in a browser to open a copy of the original workspace
1. Confirm that terminal doesn't display any warnings on patches being reapplied.

## Related Issue Link(s)

## Release/Deployment notes



<a href="https://gitpod.io/#https://github.com/shaal/DrupalPod/pull/19"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
